File: Python/Algorithms/908.py
class Solution:
    def smallestRangeI(self, A, K):
        min_ = max_ = A[0]
        for i in A:
            if i > max_:
                max_ = i
            if i < min_:
                min_ = i
        diff = max_ - K - min_
        if -K < diff < K:
            return 0
        return diff + (-K if diff > 0 else K)

File: Rust/Algorithms/908.rs
impl Solution {
    pub fn smallest_range_i(a: Vec<i32>, K: i32) -> i32 {
        let (mut min, mut max) = (a[0], a[0]);
        for i in a {
            if i < min {
                min = i;
            }
            if i > max {
                max = i;
            }
        }
        let diff = max - K - min;
        if -K < diff && diff < K {
            return 0;
        }
        if diff > 0 {
            diff - K
        } else {
            diff + K
        }
    }
}
